File: SevenWonders.java
import java.util.Scanner;

public class SevenWonders {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String input = sc.nextLine();
        
        int tCard = 0;
        int cCard = 0;
        int gCard = 0;
        int total = 0;
        int counter = 1;
        
        for(int i = 0; i < input.length(); i++) {
            
            char t = 'T';
            char g = 'G';
            char c = 'C';
            
            if(input.charAt(i) == t){
                tCard++;
            } else if(input.charAt(i) == g){
                gCard++;
            } else if(input.charAt(i) == c) {
                cCard++;
            }

            if(tCard >= counter && gCard >= counter && cCard >= counter) {
                total += 7;
                counter++;
            }
        }
    
        total += (int) (Math.pow(tCard,2) + Math.pow(gCard,2) + Math.pow(cCard, 2));
        
        System.out.println(total);
        
    }

}

File: Python Solutions/SevenWonders.py
a = input()
t,g,c = 0,0,0
sum = 0
count = 1

for i in range(len(a)):
    if a[i] == 'T':
        t+=1
    elif a[i] == 'G':
        g+=1
    else:
        c+=1

    if t >= count and g >= count and c >= count:
        sum += 7
        count+=1

sum += pow(t,2) + pow(c,2) + pow(g,2)
print(sum)